pkg/common/parameters.go

```diff
@@ -170,6 +174,21 @@ func ExtractAndDefaultParameters(parameters map[string]string, driverName string
 			if paramAvailabilityClass == ParameterRegionalHardFailoverClass {
 				p.ForceAttach = true
 			}
+		case ParameterKeyEnbaleConfidentialStorage:
+			paramEnbaleConfidentialStorage, err := ConvertStringToBool(v)
+			if err != nil {
+				return p, fmt.Errorf("parameters contain invalid value for EnbaleConfidentialStorage parameter: %w", err)
+			}
+
+			if paramEnbaleConfidentialStorage {
+				// DiskEncryptionKmsKey is needed to enable confidentialStorage
+				if val, ok := parameters[ParameterKeyDiskEncryptionKmsKey]; !ok || val == "" {
```

Comment:
Should we validate the regex of the DiskEncryptionKmsKey?

Reply:
I am conflicted on that, I did not find any function which validates the key and I am not sure if we should add another check which can break if something changes underneath. What is your opinion?

Reply:
Added regex check for the key